### ISSUE TITLE
empty file created when spliceTo (in EpollSocketChannel) used with non zero offset

### ISSUE THREAD

**Issue by catchsudheera** [netty/netty#3812]
when spliceTo (in EpollSocketChannel) used with non zero offset it doesn't write anything to the file, when zero is used it writes from the beginning of the request. No exceptions caught in the catch block too.

```
    try{
        final FileDescriptor fd = FileDescriptor.from(file);
        future = ch.spliceTo(fd, 1, 10);
    }catch (IOException e) {
        e.printStackTrace();
    }  
```

**Comment by normanmaurer** [comment]
@catchsudheera is the future failed by any chance ?

**Comment by catchsudheera** [comment]
@normanmaurer  future.isDone() and future.isSuccess() returns false even if the offset is set to zero(content written in to the file)

        try{
            final FileDescriptor fd = FileDescriptor.from(file);

            future = ch.spliceTo(fd, 1, 10);
            ch.flush();
        }catch (IOException e) {
            e.printStackTrace();
        }

        future.addListener(ChannelFutureListener.CLOSE);

               System.out.println(future.isDone());
               System.out.println(future.isSuccess());
               System.out.println(future.isCancellable());


above returns 
		false
		false
		true
 both offset zero and non zero

**Comment by normanmaurer** [comment]
@catchsudheera this means it is failed... check the exception:

future.cause().printStackTrace();

**Comment by catchsudheera** [comment]
@normanmaurer  `future.cause()` returns a null. 

`java.lang.NullPointerException
	at example.SpliceHandler.channelActive(SpliceHandler.java:68)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelActive(AbstractChannelHandlerContext.java:183)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelActive(AbstractChannelHandlerContext.java:169)
	at io.netty.channel.DefaultChannelPipeline.fireChannelActive(DefaultChannelPipeline.java:817)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:454)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.access$100(AbstractChannel.java:378)
	at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:424)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:357)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:258)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:111)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:137)
	at java.lang.Thread.run(Thread.java:744)`

**Comment by normanmaurer** [comment]
@catchsudheera can you please paste your full example ?

**Comment by catchsudheera** [comment]
Server
```
public class SpliceServer {

        public static void main(String[] args)  {
            new SpliceServer().spliceToFile();
        }
        public void spliceToFile() {

            EventLoopGroup bossGroup = new EpollEventLoopGroup();
            EventLoopGroup workerGroup = new EpollEventLoopGroup();

            ServerBootstrap bs = new ServerBootstrap();

            bs.group(bossGroup, workerGroup)
                    .channel(EpollServerSocketChannel.class)
                    .handler(new LoggingHandler(LogLevel.INFO))
                    .childHandler(new SpliceHandlerInitializer())
                    .childOption(EpollChannelOption.EPOLL_MODE, EpollMode.LEVEL_TRIGGERED);
            ChannelFuture f;
            try {
                f = bs.bind(8280).sync();
                f.channel().closeFuture().sync();
            } catch (InterruptedException e) {
                e.printStackTrace();
            }

            bossGroup.shutdownGracefully();
            workerGroup.shutdownGracefully();
        }
    }
```

Initializer

```
    public class SpliceHandlerInitializer extends ChannelInitializer<SocketChannel> {
    @Override
    protected void initChannel(SocketChannel socketChannel) throws Exception {
        ChannelPipeline p = socketChannel.pipeline();
        p.addLast(new SpliceHandler());

        }
    }
```
Handler
```
    public class SpliceHandler extends ChannelInboundHandlerAdapter {
        File file;
        volatile Channel channel;
        volatile ChannelFuture future;

        @Override
        public void channelReadComplete(ChannelHandlerContext ctx) throws InterruptedException {
            ctx.flush();
        }

        @Override
        public void channelActive(ChannelHandlerContext ctx) throws InterruptedException {

            double v = System.nanoTime();
            String filename = "file"+v+".dat";

            file = new File(filename);

            channel = ctx.channel();
            final EpollSocketChannel ch = (EpollSocketChannel) ctx.channel();

            try{
                final FileDescriptor fd = FileDescriptor.from(file);

                future = ch.spliceTo(fd, 1, 10);
                ch.flush();
            }catch (IOException e) {
                e.printStackTrace();
            }

            future.addListener(ChannelFutureListener.CLOSE);

            System.out.println(future.isDone());
            System.out.println(future.isSuccess());
            System.out.println(future.isCancellable());

            future.cause().printStackTrace();
        }

        @Override
        public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause){
                cause.printStackTrace();
                ctx.close();

        }
    }
```

**Comment by njhill** [comment]
This looks like the bug that was fixed in #9369.
